pkg/management/api/grpc/application_converter.go

```diff
@@ -0,0 +1,220 @@
+	return &OIDCConfig{
+		RedirectUris:           config.RedirectUris,
+		ResponseTypes:          oidcResponseTypesFromModel(config.ResponseTypes),
+		GrantTypes:             oidcGrantTypesFromModel(config.GrantTypes),
+		ApplicationType:        oidcApplicationTypeFromModel(config.ApplicationType),
+		ClientId:               config.ClientID,
+		ClientSecret:           config.ClientSecretString,
+		AuthMethodType:         oidcAuthMethodTypeFromModel(config.AuthMethodType),
+		PostLogoutRedirectUris: config.PostLogoutRedirectUris,
+	}
+}
+
+func oidcAppCreateToModel(app *OIDCApplicationCreate) *proj_model.Application {
+	return &proj_model.Application{
+		ObjectRoot: models.ObjectRoot{
+			ID: app.ProjectId,
+		},
+		Name: app.Name,
+		Type: proj_model.APPTYPE_OIDC,
+		OIDCConfig: &proj_model.OIDCConfig{
+			RedirectUris:           app.RedirectUris,
+			ResponseTypes:          oidcResponseTypesToModel(app.ResponseTypes),
+			GrantTypes:             oidcGrantTypesToModel(app.GrantTypes),
+			ApplicationType:        oidcApplicationTypeToModel(app.ApplicationType),
+			AuthMethodType:         oidcAuthMethodTypeToModel(app.AuthMethodType),
+			PostLogoutRedirectUris: app.PostLogoutRedirectUris,
+		},
+	}
+}
+
+func appUpdateToModel(app *ApplicationUpdate) *proj_model.Application {
+	return &proj_model.Application{
+		ObjectRoot: models.ObjectRoot{
+			ID: app.ProjectId,
+		},
+		AppID: app.Id,
+		Name:  app.Name,
+	}
+}
+
+func oidcConfigUpdateToModel(app *OIDCConfigUpdate) *proj_model.OIDCConfig {
+	return &proj_model.OIDCConfig{
+		ObjectRoot: models.ObjectRoot{
+			ID: app.ProjectId,
+		},
+		AppID:                  app.ApplicationId,
+		RedirectUris:           app.RedirectUris,
+		ResponseTypes:          oidcResponseTypesToModel(app.ResponseTypes),
+		GrantTypes:             oidcGrantTypesToModel(app.GrantTypes),
+		ApplicationType:        oidcApplicationTypeToModel(app.ApplicationType),
+		AuthMethodType:         oidcAuthMethodTypeToModel(app.AuthMethodType),
+		PostLogoutRedirectUris: app.PostLogoutRedirectUris,
+	}
+}
+
+func appStateFromModel(state proj_model.AppState) AppState {
+	switch state {
+	case proj_model.APPSTATE_ACTIVE:
+		return AppState_APPSTATE_ACTIVE
+	case proj_model.APPSTATE_INACTIVE:
+		return AppState_APPSTATE_INACTIVE
+	default:
+		return AppState_APPSTATE_UNSPECIFIED
+	}
+}
+
+func oidcResponseTypesToModel(responseTypes []OIDCResponseType) []proj_model.OIDCResponseType {
+	oidcResponseTypes := make([]proj_model.OIDCResponseType, len(responseTypes))
+
+	for i, responseType := range responseTypes {
+		switch responseType {
+		case OIDCResponseType_OIDCRESPONSETYPE_CODE:
+			oidcResponseTypes[i] = proj_model.OIDCRESPONSETYPE_CODE
+		case OIDCResponseType_OIDCRESPONSETYPE_ID_TOKEN:
+			oidcResponseTypes[i] = proj_model.OIDCRESPONSETYPE_ID_TOKEN
+		case OIDCResponseType_OIDCRESPONSETYPE_TOKEN_ID_TOKEN:
+			oidcResponseTypes[i] = proj_model.OIDCRESPONSETYPE_TOKEN_ID_TOKEN
+		}
```

Comment:
add default

Reply:
i think that doesnt make sense, it should be one of these 3 types if something else happens i dont want to do something